specification/_types/Knn.ts

```diff
@@ -35,3 +37,12 @@ export interface KnnQuery {
   /** Filters for the kNN search query */
   filter?: QueryContainer | QueryContainer[]
 }
+
+export interface QueryVectorBuilder {
```

Comment:
On the server this is actually a variant. Although `text_embedding` is currently the only implementation, we may want to add a `@variants container` (and make `text_embedding` optional) to allow for future expansion.